### ISSUE TITLE
Better support for developers who prefer URLs over the fluent API

### ISSUE THREAD

**Issue by Ndiritu** [microsoftgraph/msgraph-sdk-php#951]
Sample v1 snippet
```php

// v1.x
$response = $graphClient->createRequest('GET', '/users/userId/messages')
                        ->setReturnType(Model\Message::class)
                        ->execute();
```

Perhaps make models a separate package? This way developers can use graph core and the models package only.
(Full service lib would also take a dependency on the models)

_Developer uses native HTTP client and handles deserialization themselves_
```php

$guzzleClient = GraphClientFactory::create();
$response = $guzzleClient->request('GET', '/users/userId/messages');
//introduce a new "deserialize" method in models that accepts decoded JSON (a dictionary)?
$messages = (new MessageCollectionResponse())->deserialize(json_decode($response->getBody(), true));
```

**Comment by zengin** [comment]
cc: @nikithauc @sebastienlevert @ddyett as I hear similar thoughts floating around regarding TS/JS SDKs.

**Comment by sebastienlevert** [comment]
We are looking at this functionality from a global SDK perspective. Something that will take "time" but should get there at some point. 

JavaScript has been using this approach since day one with a `.api("/me")` approach and has been very successful (though, there was no service library). We would love to bring a similar concept to all languages, using our middleware pipeline and error management, etc. to provide this value.

In this case, models would need to be separate and you could use core + models to get to this result!

**Comment by darrelmiller** [comment]
@sebastienlevert We have had our JS/TS community asking for a fluent experience for 4 years.  Using paths is a fallback, not a preference.  It is also useful for making requests when a complete path is returned from the server in a response.  The motivation to provide a path based API has not been "because some people prefer it".  It is because in some cases our fluent API is not appropriate.  We should always consider the path based API a backup option to the strongly typed request factory experience.  Constructing URLs can be fraught with errors, especially when it comes to character encoding.  We should always consider the strongly typed experience our north star.

**Comment by Ndiritu** [comment]
Thanks for providing context @darrelmiller. I'll revisit this in case we receive relevant customer feedback.

**Comment by jdlien** [comment]
I'm trying to convert my application to the new fluent API. I specifically want to download profile photos for all the users in my organization, and was able to do individual photos using 
```php
$photoContent = $this->graphServiceClient->usersById($user->id)->photo()->content()->get()->wait();
```

But in previous versions I could make an array of requests and send them all at once like
```php
$data = $this->graph
            ->createRequest('POST', '/$batch')
            ->attachBody(json_encode(['requests' => $requests]))
            ->execute();
```

I can't for the life of me figure out how to do this now. Is this even possible with the fluent API? I am tempted to go back to the old API and lock my composer.json at msgraph-sdk-php RC4 so my old code works.

Any advice on how I could approach this would be appreciated!

**Comment by Ndiritu** [comment]
@jdlien your issue is addressed in #957
